src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -302,6 +302,32 @@ namespace winrt::TerminalApp::implementation
             // but LoadSettings skips it, so that the UI starts up faster.
             // Now that the UI is present we can do them with a less significant UX impact.
             _ProcessLazySettingsChanges();
+
+            FILETIME creationTime, exitTime, kernelTime, userTime, now;
+            if (GetThreadTimes(GetCurrentThread(), &creationTime, &exitTime, &kernelTime, &userTime))
+            {
+                static constexpr auto asInteger = [](const FILETIME& f) {
+                    ULARGE_INTEGER i;
+                    i.LowPart = f.dwLowDateTime;
+                    i.HighPart = f.dwHighDateTime;
+                    return i.QuadPart;
+                };
+                static constexpr auto asSeconds = [](uint64_t v) {
+                    return v * 1e-7f;
+                };
+
+                GetSystemTimeAsFileTime(&now);
+
+                const auto latency = asSeconds(asInteger(now) - asInteger(creationTime));

```

Comment:
I don't understand; is it a guarantee that this thread is the right one?

Reply:
The `Initialized` callback is on the main thread right? I mean after all it calls all those `_root`, etc. functions... Then yeah it's the right one.
I chose `GetThreadTimes` instead of `GetProcessTimes`, because `GetProcessTimes` is pretty bad when it comes to performance (intentionally so). To improve runtime performance each thread records its own kernel/user-mode times (because then you don't need to acquire any shared/global locks). But if you call `GetProcessTimes` it has to acquire a shared/global lock to read and accumulate all those counters. `GetThreadTimes` doesn't have that issue and delivers the same result if called on the main thread.